### PR TITLE
Fix MTL suspend/resume callbacks

### PR DESCRIPTION
**Pull request by ranj063**

### CODE REVIEW

File: sound/soc/sof/intel/hda-dsp.c [hunk 1]
```diff
@@ -629,10 +629,11 @@ static int hda_suspend(struct snd_sof_dev *sdev, bool runtime_suspend)
 	    sdev->fw_state == SOF_FW_BOOT_FAILED)
 		hda->skip_imr_boot = true;
 
-	hda_sdw_int_enable(sdev, false);
-
-	/* disable IPC interrupts */
-	hda_dsp_ipc_int_disable(sdev);
+	ret = chip->disable_interrupts(sdev);
+	if (ret < 0) {
+		dev_err(sdev->dev, "failed to disable DSP interrupts during suspend\n");
```
Comment (RanderWang) [inline sound/soc/sof/intel/hda-dsp.c]: print ret in log

**Review by RanderWang — COMMENTED** [review]
LGTM

File: sound/soc/sof/intel/hda-dsp.c (hunk 1 above)
Comment (ranj063) [reply sound/soc/sof/intel/hda-dsp.c]: is this really helpful? if you look at the HDA ops, they never return an error. And as for the MTL op, it prints the error message for each condition that fails already no?

Comment (RanderWang) [reply sound/soc/sof/intel/hda-dsp.c]: if error has been print, we can just return ret without error message. I don't want many error msg.

**Review by bardliao — APPROVED** [review]

File: sound/soc/sof/intel/hda-dsp.c (hunk 1 above)
Comment (ranj063) [reply sound/soc/sof/intel/hda-dsp.c]: yeah thats what i was thinking too. fixed now

**Review by bardliao — APPROVED** [review]

**Review by RanderWang — APPROVED** [review]

**Review by plbossart — APPROVED** [review]

**Pull request merged**